packages/api/src/SubmittableExtrinsic.ts

```diff
@@ -197,19 +197,37 @@ export default function createSubmittableExtrinsic<ApiType> (
 
           return decorateMethod(
             () => ((
-              isUndefined(options.nonce)
-                ? api.query.system.accountNonce<Index>(address)
-                : of(new Index(options.nonce))
+              combineLatest([
+                isUndefined(options.nonce)
+                  ? api.query.system.accountNonce<Index>(address)
+                  : of(new Index(options.nonce)),
+                isUndefined(options.era) && options.eraLength !== -1
+                  ? api.rpc.chain.getBlock() as Observable<SignedBlock>
+                  : of(null)
+              ])
             ).pipe(
               first(),
-              mergeMap(async (nonce) => {
+              mergeMap(async ([nonce, signedBlock]) => {
+                let blockHash: Hash | undefined;
+                let era: ExtrinsicEra | undefined;
+
+                if (signedBlock) {
```

Comment:
In which case would signedBlock be falsy?

Reply:
when either -

- era is set, or
- eraLength is set to something that is not -1

Reply:
Added an explicit comment around the of as to why

Reply:
So what I dislike about this, it's that
1/ this -1 has hidden meaning, i.e. you need to read docs to know about it
2/ the api user won't know what happens at a glance when I do
`signAndSend(..., { eraLength: -1, era: new Era('something mortal')}) // which one takes precedence?`
or even 
`signAndSend(..., { eraLength: 10, era: new Era({ ..., period: 20 })}) // which one takes precedence?`

I would remove eraLength, and change type of era to e.g.
```
type era = 'Immortal' | Era | AnyNumber;
```

So that 
```
signAndSend(..., { era: 'Immortal' });
signAndSend(..., { era: new Era('the immortal enum') });
signAndSend(..., { era: new Era('some mortal era') });
signAndSend(..., { era: 10 }); // should be clear that here the 10 means era length, esp if the user passes blockHash too
```

Reply:
Cool, yes, that is sane - I didn't like the `eraLength` as well as `era` definitions. (And at the same time didn't want to remove `era` completely)

Adjusted.